### PR TITLE
Fix c++20 incompatibility

### LINKED ISSUE #406: C++20 support
**Description**

```
../apps/websocket_node/foxglove_websocket/include/foxglove/websocket/websocket_server.hpp:266:37: error: aggregate initialization of type 'foxglove::Server<foxglove::WebSocketNoTls>::ClientInfo' with user-declared constructors is incompatible with C++20 [-Werror,-Wc++20-compat]
    _clients.emplace(hdl, ClientInfo{endpoint, hdl, {}, {}});
                                    ^~~~~~~~~~~~~~~~~~~~~~~
```

### PR DESCRIPTION
**Pull request by achim-k**
### Public-Facing Changes

- Fix c++20 incompatibility

### Description
Fixes #406

### CODE REVIEW

**Review by jtbandes — APPROVED** [review]
Can we add a C++20 build to CI so we can catch future issues?

**Comment by defunctzombie** [comment]
image

Note the build passed but not really :D

File: cpp/Makefile [hunk 1]
```diff
@@ -2,7 +2,11 @@ default: build
 
 .PHONY: build
 build:
-	docker compose build
+	docker compose build --build-arg CPPSTD=17
+
+.PHONY: build-cpp20
+build:
```
Comment (defunctzombie) [inline cpp/Makefile:8]:
```suggestion
build-cpp20:
```

Comment (achim-k) [reply cpp/Makefile]: Thanks! :see_no_evil:

**Comment by defunctzombie** [comment]
LGTM - tho you could decrease the end-to-end CI time by running the two builds in parallel if you want (future enhancement)

**Pull request merged**